proposals/0091-SDLScreen-SDLWindow-Projection.md

```diff
@@ -1,110 +1,209 @@
+// Update focus regions whenever the layout has changed
+// https://developer.apple.com/documentation/quartzcore/calayer/1410935-layoutsublayers
+//==========================================================================================
+static void (*pfnCALayer_layoutSublayers)(CALayer*, SEL) = nil;
+static void CALayer_layoutSublayers(CALayer* layer, SEL methodName)
+{
+    if (pfnCALayer_layoutSublayers) {
+        pfnCALayer_layoutSublayers(layer, methodName);
+    }
+    SDLCarWindow* carWindowForLayer = objc_getAssociatedObject(layer, KEY_CARWINDOW);
+    if (([carWindowForLayer isKindOfClass:[SDLCarWindow class]] == YES) &&
+        ([carWindowForLayer.sdlLayer isEqual:layer] == YES)) {
+        [carWindowForLayer updateInterfaceLayout];
+    }
+}
 ```
+## Benefits
+* The projected video view is separate from the view on the device's LCD allowing for optional use of device while projecting.
+* Video projection is achieved using documented Apple procedures for external display support. 
+* The app developer doesn't need to write any code to handle touch events when using supported UIKit responders. From the app's perpective, actions are triggered identically to UIKit.
+* The app developer has the option to handle touch events in the app should they wish to do so. (Custom views, OpenGL)
+* Touch event behavior of the iOS proxy remains consistent across all apps using the same version of the SDL framework. 
+* VPM features and fixes propogate to all apps.
 
 ## Potential downsides
-
-In choosing a projection API that differs from the standard UIScreen/UIWindow model for displaying content on an external screen, we are asking developers to adopt a model that is somewhat unfamiliar. However, it is also similar enough that the learning curve should be short and it avoids the possibility of strange side-effects due to subclassing UIKit controls.
+* SDLCarWindow uses swizzling. All swizzled methods are public and have not changed since iOS 2.0-3.2.
+* Unforeseen challenges may arise when adding support for new UIKit types. 
 
 ## Impact on existing code
-
-This proposal has no impact on existing video/audio streaming applications. It provides a new higher level interface that should improve developer productivity and a foundation for both touch and physical button focus/select interactions.
+* Apps adopting SDLTouchManagerDelegate will need to return BOOL instead of void for delegate methods.
+* SDLTouchManager moves from SDLStreamingMediaLifecycleManager to SDLCarWindow
+* SDLCarWindow deprecates SDLHapticManager. Functionality previously found in SDLHapticManager has been moved into SDLCarWindow.  
 
 ## Alternatives considered
+* Force the each application to handle touch events via SDLTouchManagerDelegate. The developer has to handle the translation of a view and location into the desired action. Touch management will be inconsist amongst SDL apps, which could lead to user dismay. 
```

Comment:
I believe "inconsist" should be updated to "inconsistent".

Reply:
We've decided it would be best to move touch manager related changes to another proposal. This entire paragraph goes away.